akka-http-core/src/main/scala/akka/http/engine/client/HostConnectionPoolConductor.scala

```diff
@@ -0,0 +1,230 @@
+  import HostConnectionPoolSlot.{ SlotEvent, SimpleSlotEvent }
+
+  case class Ports(
+    requestIn: Inlet[RequestContext],
+    slotEventIn: Inlet[SlotEvent],
+    slotOuts: immutable.Seq[Outlet[RequestContext]]) extends Shape {
+
+    override val inlets = requestIn :: slotEventIn :: Nil
+    override def outlets = slotOuts
+
+    override def deepCopy(): Shape =
+      Ports(
+        new Inlet(requestIn.toString),
+        new Inlet(slotEventIn.toString),
+        slotOuts.map(o ⇒ new Outlet(o.toString)))
+
+    override def copyFromPorts(inlets: immutable.Seq[Inlet[_]], outlets: immutable.Seq[Outlet[_]]): Shape =
+      Ports(
+        inlets.head.asInstanceOf[Inlet[RequestContext]],
+        inlets.last.asInstanceOf[Inlet[SlotEvent]],
+        outlets.asInstanceOf[immutable.Seq[Outlet[RequestContext]]])
+  }
+
+  /*
+    Stream Setup
+    ============
+                                                                                                  Request- 
+    Request-   +-----------+     +-----------+    Switch-    +-------------+     +-----------+    Context
+    Context    |   merge1  |     |   merge2  |    Command    |   doubler   |     |   route   +-------------->
+    +--------->|  (Merge-  +---->|  (Flexi   +-------------->| (MapConcat) +---->|  (Flexi   +-------------->
+               | Preferred)|     |   Merge)  |               |             |     |   Route)  +-------------->
+               +----+------+     +-----+-----+               +-------------+     +-----------+       to slots     
+                    ^                  ^                                                               
+                    |                  | SimpleSlotEvent                                              
+                    | Request-         |                                                               
+                    | Context     +---------+
+                    +-------------+  retry  |<-------- Slot Event (from slotEventMerge)
+                                  |  Split  |
+                                  +---------+
+
+  */
+  def apply(slotCount: Int, maxRetries: Int, pipeliningLimit: Int, log: LoggingAdapter): Graph[Ports, Any] =
+    FlowGraph.partial() { implicit b ⇒
+      import FlowGraph.Implicits._
+
+      val merge1 = b.add(new Merge1)
+      val merge2 = b.add(new Merge2(slotCount, maxRetries, pipeliningLimit, log))
+      val doubler = Flow[SwitchCommand].mapConcat(x ⇒ x :: x :: Nil)
+      val route = b.add(new Route(slotCount))
+      val retrySplit = b.add(new RetrySplit())
+
+      merge1.out ~> merge2.in0
+      merge2.out ~> doubler ~> route.in
+      retrySplit.out0 ~> merge2.in1
+      retrySplit.out1 ~> merge1.in1
+
+      Ports(merge1.in0, retrySplit.in, route.outlets.asInstanceOf[immutable.Seq[Outlet[RequestContext]]])
+    }
+
+  // almost the same as a MergePreferred,
+  // but as MergePreferred doesn't appear to propagate completion on the secondary input we can use it here
```

Comment:
but this does not prefer any inputs? You are only interested in the eager close right?